.github/workflows/plugin-deploy.yml

```diff
@@ -0,0 +1,61 @@
+name: Deploy plugin to Azure function app
+
+on:
+  workflow_call:
+    secrets:
+      PYTHON_VERSION:
+        required: true
+      AZURE_CREDENTIALS:
+        required: true
+      AZURE_FUNCTIONAPP_NAME:
+        required: true
+      AZURE_FUNCTIONAPP_PUBLISH_PROFILE:
+        required: true
+      GIT_TOKEN:
+        required: true
+      GIT_URL_PLUGIN_DETAILS:
+        required: true
+      AZURE_FUNCTIONAPP_PACKAGE_PATH:
+        required: true
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: checkout repo
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          token: ${{ secrets.GIT_TOKEN }}
+      - name: "Login via Azure CLI"
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: setup Python ${{ secrets.PYTHON_VERSION }} environment
```

Comment:
Should we not hardcode the py version as we want it to be a specific version for all plugins at this stage?

Reply:
what do you mean by "hardcode" here? I made the python version a secret precisely so we can have a different version for each plugin, if we need to.